main.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    counter = 0
+    while counter < size
+      yield(self[counter])
+      counter += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    output = []
+    my_each { |ele| output << ele if yield(ele) }
+    output
+  end
+
+  def my_all?
+    output = true
+    my_each do |ele|
+      output = false if yield(ele) != true
+    end
+    output
+  end
+
+  def my_any?
+    output = false
+    my_each do |ele|
+      output = true if yield(ele) == true
+    end
+    output
+  end
+
+  def my_none?
```

Comment:
#### `REQUIRED`

### :bug: Let's add a case for when a block is not given. Compare with the built-in method. Learn more [here](https://ruby-doc.org/core-2.6.3/Enumerable.html#method-i-none-3F)

#### 1. When `arg` is not a `block`

```ruby
irb(main):005:0> [1,2,3,4,5].none?
=> false
irb(main):006:0> [1,2,3,4,5].my_none?
Traceback (most recent call last):
        7: from /../.rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        6: from /../.rbenv/versions/2.6.3/bin/irb:23:in `load'
        5: from /../.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        4: from (irb):6
        3: from /../enumerable/main.rb:44:in `my_none?'
        2: from /../enumerable/main.rb:7:in `my_each'
        1: from /../enumerable/main.rb:45:in `block in my_none?'
LocalJumpError (no block given (yield))
```

#### 2. When a `block` is given (This have to do with the `puts` used)

```ruby
irb(main):007:0> [1,2,3,4,5].none?{|i| i > 2}
=> false
irb(main):008:0> [1,2,3,4,5].my_none?{|i| i > 2}
false
=> nil
```

#### 3. When `arg` is not a `block`

```ruby
irb(main):010:0> [1,2,3,4,5].none?(3)
=> false
irb(main):011:0> [1,2,3,4,5].my_none?(3)
Traceback (most recent call last):
        5: from /../.rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        4: from /../.rbenv/versions/2.6.3/bin/irb:23:in `load'
        3: from /../.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):11
        1: from /../enumerable/main.rb:42:in `my_none?'
ArgumentError (wrong number of arguments (given 1, expected 0))

irb(main):013:0> [1,2,3,4,5].none?(7)
=> true
irb(main):014:0> [1,2,3,4,5].my_none?(7)
Traceback (most recent call last):
        5: from /../.rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        4: from /../.rbenv/versions/2.6.3/bin/irb:23:in `load'
        3: from /../.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):14
        1: from /../enumerable/main.rb:42:in `my_none?'
ArgumentError (wrong number of arguments (given 1, expected 0))
```